client/asset/btc/btc.go

```diff
@@ -2133,6 +2214,34 @@ func (btc *ExchangeWallet) getVersion() (uint64, uint64, error) {
 	return r.Version, r.ProtocolVersion, nil
 }
 
+// externalAddress will return a new address for public use.
+func (btc *ExchangeWallet) externalAddress() (btcutil.Address, error) {
+	if btc.segwit {
+		return btc.wallet.AddressWPKH()
+	}
+	return btc.wallet.AddressPKH()
+}
+
+// hashContract hashes the contract. The hash function used depends on whether
+// the wallet is configured for segwit.
+func (btc *ExchangeWallet) hashContract(contract []byte) []byte {
+	if btc.segwit {
+		h := sha256.Sum256(contract)
+		return h[:]
+	}
+	return btcutil.Hash160(contract)
```

Comment:
Can we comment the intent like

```go
	h := sha256.Sum256(contract) // bip141
...
return btcutil.Hash160(contract) // bip16
```

Esp. since the function `hashContract` isn't indicating that the hash is for a pkScript